fsw/src/bpnode_cmds.c

```diff
@@ -0,0 +1,104 @@
+/************************************************************************
+ * NASA Docket No. GSC-18,719-1, and identified as “core Flight System: Bootes”
+ *
+ * Copyright (c) 2020 United States Government as represented by the
+ * Administrator of the National Aeronautics and Space Administration.
+ * All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain
+ * a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ************************************************************************/
+
+/**
+ * \file
+ *   This file contains the source code for the BPNode command-handling functions
+ */
+
+/*
+** Include Files:
+*/
+#include "bpnode_app.h"
+#include "bpnode_cmds.h"
+#include "bpnode_msgids.h"
+#include "bpnode_eventids.h"
+#include "bpnode_version.h"
+#include "bpnode_tbl.h"
+#include "bpnode_utils.h"
+#include "bpnode_msg.h"
+
+#include "bplib.h"
+
+/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * **/
+/*                                                                            */
+/*  Purpose:                                                                  */
+/*         This function is triggered in response to a task telemetry request */
+/*         from the housekeeping task. This function will gather the Apps     */
+/*         telemetry, packetize it and send it to the housekeeping task via   */
+/*         the software bus                                                   */
+/* * * * * * * * * * * * * * * * * * * * * * * *  * * * * * * *  * *  * * * * */
+CFE_Status_t BPNode_SendHkCmd(const BPNode_SendHkCmd_t *Msg)
+{
+    int i;
+
+    /*
+    ** Get command execution counters...
+    */
+    BPNode_AppData.HkTlm.Payload.CommandErrorCounter = BPNode_AppData.ErrCounter;
+    BPNode_AppData.HkTlm.Payload.CommandCounter      = BPNode_AppData.CmdCounter;
+
+    /*
+    ** Send housekeeping telemetry packet...
+    */
+    CFE_SB_TimeStampMsg(CFE_MSG_PTR(BPNode_AppData.HkTlm.TelemetryHeader));
+    CFE_SB_TransmitMsg(CFE_MSG_PTR(BPNode_AppData.HkTlm.TelemetryHeader), true);
+
+    /*
+    ** Manage any pending table loads, validations, etc.
+    */
+    for (i = 0; i < BPNODE_NUMBER_OF_TABLES; i++)
+    {
+        CFE_TBL_Manage(BPNode_AppData.TblHandles[i]);
+    }
```

Comment:
Can we move this to the wakeup?

Reply:
Will do

Reply:
Done

Reply:
Oh, something I didn't catch here before - I think we're supposed to do a 3-step process here to update our tables.

- CFE_TBL_ReleaseAddress
- CFE_TBL_Manage
- CFE_TBL_GetAddress

Can you check whether this is true, and if so, update this?

Reply:
Yep, that's the right process. Updated things, should be good now